soh/soh/Enhancements/cosmetics/CosmeticsEditor.cpp

```diff
@@ -400,6 +402,10 @@ void CosmeticsUpdateTick() {
             newColor.g = sin(frequency * (hue + index) + (2 * M_PI / 3)) * 127 + 128;
             newColor.b = sin(frequency * (hue + index) + (4 * M_PI / 3)) * 127 + 128;
             newColor.a = 255;
+            // For alpha supported options, retain the last set alpha instead of overwriting
+            if (cosmeticOption.supportsAlpha) {
+                newColor.a = cosmeticOption.currentColor.w * 255;
+            }
```

Comment:
Since this PR enables support for alpha control in the cosmetic editor (on supported options, currently only the health bar), I decided that alpha should not be change for rainbow and randomize.

Randomizing alpha seems like a weird idea. My thinking is that the user can choose their own alpha (i.e. a 50% opacity health bar), and then they can still randomize/rainbow it while keeping the alpha they chose.

Looking for feedback here on this decision.

Reply:
yeah, randomizing alpha feels odd to me, i'm on board with this decision